graphql/parser.go

```diff
@@ -36,7 +36,7 @@ func valueToJson(value ast.Value, vars map[string]interface{}) (interface{}, err
 	case *ast.Variable:
 		actual, ok := vars[value.Name.Value]
 		if !ok {
-			return nil, NewClientError("unknown var: %s", value.Name.Value)
+			return nil, nil
```

Comment:
let's add tests for the 2x2:
- argument is optional, and argument is not specified in variables
- argument is optional, and argument is specified in variables
- argument is not optional, and argument is not specified in variables
- argument is not optional, and argument is specified in variables